src/org/openstreetmap/josm/actions/mapmode/SplitMode.java

```diff
@@ -0,0 +1,352 @@
+    HighlightHelper highlight = new HighlightHelper();
+
+    /**
+     * Construct a new SplitMode object
+     * @since XXX
+     */
+    public SplitMode() {
+        super(tr("Split mode"), "splitway", tr("Split ways"),
+        Shortcut.registerShortcut("mapmode:split", tr("Mode: {0}", tr("Split mode")), KeyEvent.VK_T, Shortcut.DIRECT),
+        ImageProvider.getCursor("crosshair", null));
+    }
+    
+    @Override
+    public void enterMode() {
+        super.enterMode();
+        MapView mv = MainApplication.getMap().mapView;
+        mv.addMouseListener(this);
+        mv.addMouseMotionListener(this);
+    }
+
+    @Override
+    public void exitMode() {
+        super.exitMode();
+        MapView mv = MainApplication.getMap().mapView;
+        mv.removeMouseMotionListener(this);
+        mv.removeMouseListener(this);
+        removeHighlighting();
+    }
+
+    @Override
+    public void mousePressed(MouseEvent e) {
+        super.mousePressed(e);
+
+        MapView mv = MainApplication.getMap().mapView;
+        int mouseDownButton = e.getButton();
+        Point mousePos = e.getPoint();
+
+        // return early
+        if (!mv.isActiveLayerVisible() || Boolean.FALSE.equals(this.getValue("active")) || mouseDownButton != MouseEvent.BUTTON1)
+            return;
+
+        // update which modifiers are pressed (shift, alt, ctrl)
+        updateKeyModifiers(e);
+
+        DataSet ds = getLayerManager().getEditDataSet();
+        if (ds == null)
+            return;
+
+        final List<Way> selectedWays = new ArrayList<>(ds.getSelectedWays());
+        Optional<OsmPrimitive> primitiveAtPoint = getPrimitiveAtPoint(e.getPoint());
+        if (!primitiveAtPoint.isPresent())
+            return;
+            
+        final OsmPrimitive nearestPrimitive = primitiveAtPoint.get();
+        
+        if (nearestPrimitive instanceof Node) {
+            // Split way at node
+            Node n = (Node) nearestPrimitive;
+
+            List<Way> applicableWays = getApplicableWays(n, selectedWays);
+
+            if (applicableWays.isEmpty()) {
+                new Notification(
+                        tr("The selected node is not in the middle of any non-closed way."))
+                        .setIcon(JOptionPane.WARNING_MESSAGE)
+                        .show();
+                return;
+            }
+
+            if (applicableWays.size() > 1) {
+                createPopup(n, applicableWays).show(mv, mousePos.x, mousePos.y);
+                return;
+            } else {
+                final Way splitWay = applicableWays.get(0);
+                SplitWayAction.doSplitWayShowSegmentSelection(splitWay, Collections.singletonList(n), null);
+                if (updateUserFeedback(e)) {
+                    MainApplication.getMap().mapView.repaint();
+                }
+            }
+        } else if (nearestPrimitive instanceof Way) {
+            // TODO: Implement way splitting when a point on a way segment (not a node) is selected
```

Comment:
Unfortunately it's not quite as easy as that. The case with only one way is fairly trivial, that is true, but once multiple overlapping ways are involved it gets a bit more complicated. 

The process essentially boils down to:
1. Create a new node and join it to way(s)
2. Split way(s) at new node

We don't want to do this individually one-by-one for each way. Adding the new node to a single way only when there are multiple overlapping ways sharing the same way segments would 'unjoin' that segment from the rest for that particular way. That means we probably need to add the new node to all ways sharing the same way segment regardless of whether they should be split or not. (Which also leads to further considerations that we need to think about such as cases where some of those ways are disabled/hidden by a filter, etc...)

Step 1. is trivial for one way, but quickly gets messy for multiple ways as evidenced by this mess https://github.com/JOSM/josm/blob/3af7bae967e273dcee423ca0aac04524615d5ba5/src/org/openstreetmap/josm/actions/JoinNodeWayAction.java#L90 (although that code is a lot less elegant than it could be and also needs to handle joining multiple nodes at once which leads to more edge cases to handle). I am actually in the process of refactoring that code so that parts can be reused for the splitting here.

Step 2. is more of a problem of allowing the user to choose which ways to split, but that's pretty similar to what already exists for splitting at nodes. Splitting of multiple ways simultaneously would be nice to have, but it's not absolutely necessary. After the first way is split in the middle of a way segment, the split-at-node function can be used for the rest since the newly created node is also added to the other non-split ways.